src/Aspire.Hosting/ApplicationModel/ResourceNotificationService.cs

```diff
@@ -389,6 +390,20 @@ public Task PublishUpdateAsync(IResource resource, string resourceId, Func<Custo
         return Task.CompletedTask;
     }
 
+    /// <summary>
+    /// Update resource snapshot health status if the resource is running with no health checks.
+    /// </summary>
+    private static CustomResourceSnapshot UpdateHealthStatus(IResource resource, CustomResourceSnapshot previousState)
+    {
+        // A resource is also healthy if it has no health check annotations and is in the running state.
+        if (!resource.TryGetAnnotationsIncludingAncestorsOfType<HealthCheckAnnotation>(out _) && previousState.State?.Text == KnownResourceStates.Running)
```

Comment:
I don't think this is sufficient. If you look at `ResourceHealthCheckService` it considers the health checks of parent resources when determining child resource health. What we have here could introduce subtle bugs around child resource status.

We are better off leaving the true health status to be determined by the `ResourceHealthCheckService`.

Reply:
Actually upon reflection this is fine. You aren't interpreting health status here you are just asserting that none of the resources or their parents have health check annotations.

Coincidently this should be rare because in most cases where we have a parent and a child resource we have implemented health checks in the parent at least.

Reply:
Should there be tests around parent/child resources and health status?

Reply:
I wouldn't think so, but @mitchdenny?

Reply:
It would be good to have some from a dashboard perspective. We have the health status itself covered on the app model side of things.